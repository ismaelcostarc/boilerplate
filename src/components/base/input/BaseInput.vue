<script setup lang="ts">
  const props = defineProps<{
    placeholder?: string
    modelValue: string
  }>()

  const emit = defineEmits(['update:modelValue'])

  const updateValue = (event: Event) => {
    emit('update:modelValue', (event.target as HTMLInputElement).value)
  }
</script>

<template>
  <input
    type="text"
    class="border border-solid border-gray-400 p-2 rounded-md"
    :placeholder="props.placeholder"
    :value="props.modelValue"
    @input="updateValue"
  />
</template>
