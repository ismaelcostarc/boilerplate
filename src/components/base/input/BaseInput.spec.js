import { mount } from '@vue/test-utils'
import BaseInput from './BaseInput.vue'
import { describe, it, expect, beforeEach } from 'vitest'

describe('BaseInput', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(BaseInput, {
      props: {
        type: 'bg-white',
      },
      slots: {
        default: 'Content',
      },
    })
  })

  it.skip('should mount component', () => {
    expect(BaseInput).toBeTruthy()

    expect(wrapper.text()).toContain('Content')
  })

  it.skip('should be clicked', async () => {
    expect(BaseInput).toBeTruthy()

    await wrapper.get('button').trigger('click')

    expect(wrapper.emitted()).toHaveProperty('click')
  })

  it.skip('should match snapshot UI testing', () => {
    expect(wrapper.text()).toMatchSnapshot()
  })
})
