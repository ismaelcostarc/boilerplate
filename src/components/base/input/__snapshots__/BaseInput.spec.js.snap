// Vitest Snapshot v1

exports[`BaseInput > should match snapshot UI testing 1`] = `""`;
